src/Microsoft.OData.Client/ALinq/QueryableResourceExpression.cs

```diff
@@ -371,6 +371,56 @@ internal void AddFilter(IEnumerable<Expression> predicateConjuncts)
             this.keyPredicateConjuncts.Clear();
         }
 
+        internal void AddApply(Expression aggregateExpr, OData.UriParser.Aggregation.AggregationMethod aggregationMethod)
+        {
+            if (this.OrderBy != null)
+            {
+                throw new NotSupportedException(Strings.ALinq_QueryOptionOutOfOrder("apply", "orderby"));
+            }
+            else if (this.Skip != null)
+            {
+                // $skip and $top may be used together with rollup. 
+                // However, support for rollup is currently not implemented in OData WebApi
+                // If $skip and/or $top appears before $apply, its currently ignored. 
+                // Makes sense to throw an exception to avoid giving a false impression.
+                throw new NotSupportedException(Strings.ALinq_QueryOptionOutOfOrder("apply", "skip"));

```

Comment:
@habbes I'll sidebar with you on this

Reply:
@habbes @odero Just to clarify, ODL doesn't support `rollup` either.
The `rollup` grouping operator allows requesting additional levels of aggregation in addition to the most granular level defined by the grouping properties. It can be used instead of a property path in the first parameter of `groupby`. To me its a feature for the future if ever.

Here's a link to the section of the spec that describes how it works http://docs.oasis-open.org/odata/odata-data-aggregation-ext/v4.0/cs02/odata-data-aggregation-ext-v4.0-cs02.html#_Toc435016584